app/code/Magento/Indexer/Console/Command/IndexerStatusMviewCommand.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Indexer\Console\Command;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Command\Command;
+use Magento\Framework\Mview\View;
+use Magento\Framework\Mview\View\CollectionFactory;
+use Magento\Framework\Console\Cli;
+
+/**
+ * Command for displaying status of mview indexers.
+ */
+class IndexerStatusMviewCommand extends Command
+{
+    /** @var \Magento\Framework\Mview\View\CollectionInterface $mviewCollection */
+    private $mviewCollection;
+
+    public function __construct(
+        CollectionFactory $collectionFactory
+    ) {
+        $this->mviewCollection = $collectionFactory->create();
+
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setName('indexer:status:mview')
+            ->setDescription('Shows status of Mview Indexers and their queue status');
+
+        parent::configure();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        try {
+            $table = $this->getHelperSet()->get('table');
+            $table->setHeaders(['ID', 'Mode', 'Status', 'Updated', 'Version ID', 'Backlog']);
+
+            $rows = [];
+
+            /** @var \Magento\Framework\Mview\View $view */
+            foreach ($this->mviewCollection as $view) {
+                $state = $view->getState();
+                $changelog = $view->getChangelog();
+
+                try {
+                    $currentVersionId = $changelog->getVersion();
+                } catch (View\ChangelogTableNotExistsException $e) {
+                    continue;
+                }
+
+                $pendingCount = count($changelog->getList($state->getVersionId(), $currentVersionId));
```

Comment:
I believe that this code will have memory usage  issue in case of huge number of item in log

Reply:
Good point, it's defined as 

```php
    public function getList($fromVersionId, $toVersionId)
    {
        $changelogTableName = $this->resource->getTableName($this->getName());
        if (!$this->connection->isTableExists($changelogTableName)) {
            throw new ChangelogTableNotExistsException(new Phrase("Table %1 does not exist", [$changelogTableName]));
        }

        $select = $this->connection->select()->distinct(
            true
        )->from(
            $changelogTableName,
            [$this->getColumnName()]
        )->where(
            'version_id > ?',
            (int)$fromVersionId
        )->where(
            'version_id <= ?',
            (int)$toVersionId
        );

        return $this->connection->fetchCol($select);
    }
```

I'll address.